src/test/java/org/opensearch/sdk/sample/helloworld/TestHelloWorldExtension.java

```diff
@@ -245,11 +245,8 @@ public void onFailure(Exception e) {
         assertEquals("failed to find action [" + UnregisteredAction.INSTANCE + "] to execute", ex.getMessage());
     }
 
+    @Test
     public void testValidatedSettings() {
-        final String expected = randomAlphaOfLengthBetween(1, 5);
-        final String actual = VALIDATED_SETTING.get(Settings.builder().put(VALIDATED_SETTING.getKey(), expected).build());
-        assertEquals(expected, actual);
```

Comment:
You're actually removing the test case here that a setting *not* vailing validation doesn't throw an exception.  So this part needs to stay, just change your expected away from a random alpha to a string you choose.

Also I don't see `VALIDATED_SETTING` being updated to use the new boolean.  We would want to create two settings, one with the forbidden regex (this one) and one with a "must match" regex (I would suggest an IP address matcher, or file path matcher, or URL matcher, or any of the simpler matches here:  https://digitalfortress.tech/tips/top-15-commonly-used-regex/